### PR TITLE
Repo Sync through GitHub actions

### LINKED ISSUE #114: Conflicting filenames on fresh clone
Upon creating a fresh clone of the repository, I am receiving the following error:
```
Cloning into 'hiop'...
remote: Enumerating objects: 419, done.
remote: Counting objects: 100% (419/419), done.
remote: Compressing objects: 100% (228/228), done.
remote: Total 5532 (delta 276), reused 280 (delta 190), pack-reused 5113
Receiving objects: 100% (5532/5532), 31.97 MiB | 7.78 MiB/s, done.
Resolving deltas: 100% (4140/4140), done.
warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree:

  'BUILD.sh'
  'build.sh'
```

I think this can be fixed by simply renaming one or the other or removing one or the other assuming they were the same, which doesn't seem to be the case. Otherwise, depending on the system configuration cloning the repo, you will get mixed results.

Tried setting global git config to ignore case but didn't help as expected.

I'll poke at a PR to address this.

### PR DESCRIPTION
**Pull request by GitBytes**
Adds github action for syncing repo to gitlab.pnnl.gov

### CODE REVIEW

**Comment by pelesh** [comment]
@ashermancinelli, on some platforms `BUILD.sh` and `build.sh` conflict with each other; this is probably what tripped @GitBytes. Do we still need `build.sh`?

**Comment by GitBytes** [comment]
@ashermancinelli @pelesh If you decide we only need one build file I can move forward on pulling one or the other out via this PR as well. Thanks.

**Comment by ashermancinelli** [comment]
> @ashermancinelli, on some platforms `BUILD.sh` and `build.sh` conflict with each other; this is probably what tripped @GitBytes. Do we still need `build.sh`?

No, we do not. I've opened #118 to track this so I think we should remove the renaming from the diff and merge this.

**Review by ashermancinelli — CHANGES_REQUESTED** [review]
Approve after renaming of build script is removed.

**Comment by GitBytes** [comment]
@ashermancinelli I removed the rename in relation to #114 per your recommended changes. #114 is related to #118 just FYI.

**Review by cnpetra — APPROVED** [review]

**Review by ashermancinelli — APPROVED** [review]
LGTM. Thank you for setting this up @GitBytes!

File: .github/workflows/push_mirror.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  to_gitlab:
+    runs-on: ubuntu-18.04
+    steps:                                              # <-- must use actions/checkout@v1 before mirroring!
+    - uses: actions/checkout@v1
+    - uses: pixta-dev/repository-mirroring-action@v1
+      with:
+        target_repo_url:
+          git@gitlab.pnnl.gov:2222/exasgd/solvers/hiop.git
```
Comment (pelesh) [inline .github/workflows/push_mirror.yml:13]:
```suggestion
          git@gitlab.pnnl.gov:2222/exasgd/solvers/hiop-mirror.git
```
This should go to HiOp mirror repository at our GitLab.

Comment (pelesh) [reply .github/workflows/push_mirror.yml:13]: Otherwise looks good.

Comment (GitBytes) [reply .github/workflows/push_mirror.yml:13]: Oh.. ok interesting I thought we might just want to push this to the existing code repo (non `mirror` variant). So you want to maintain that `-mirror` variant. Understood. I'll make the change.

Comment (GitBytes) [reply .github/workflows/push_mirror.yml]: Done, ready for review @pelesh

**Review by pelesh — APPROVED** [review]

**Pull request merged**